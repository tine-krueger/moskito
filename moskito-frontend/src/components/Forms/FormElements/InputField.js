import styled from 'styled-components/macro'
import PropTypes from 'prop-types'

InputField.propTypes = {
    type: PropTypes.string.isRequired,
    name: PropTypes.string,
    value: PropTypes.node,
    onChange: PropTypes.func,
    placeholder: PropTypes.node,
    children: PropTypes.node,
    marginBottom: PropTypes.string,
    min: PropTypes.number,
    max: PropTypes.number,
    autocomplete: PropTypes.string
}

export default function InputField({
    type, 
    name, 
    value, 
    onChange, 
    placeholder, 
    children, 
    marginBottom, 
    min, 
    max, 
    autocomplete
}) {

    return (
        <Label data-testid='input-field' lessMargin={marginBottom}>
            <h3>{children}</h3>
            <input 
            data-testid={name} 
            type={type} 
            name={name} 
            value={value} 
            onChange={onChange} 
            placeholder={placeholder} 
            min={min} 
            max={max} 
            autoComplete={autocomplete} 
            />
        </Label>
    )
}

const Label = styled.label `
    grid-column: 1 / -1;
    width: 100%;
    text-align: center;
    margin: 0 auto 2em;
    margin-bottom: ${props => props.lessMargin === 'sm' ? '.5em' : '1em'};
    h3 {
        text-align: left;
        padding-left: .5em;
        margin: 0 0 1em;
    }

    input {
        width: 100%;
        padding: 1em;
        height: 3em;
        font-size: 1em;
        border: none;
        border-radius: 10px;
        color: #6b717e;
        background-color: #d8e6e4;
        box-shadow: inset 4px 4px 6px 0 rgba(0,0,0,0.2), inset -3px -4px 6px 0 rgba(255,255,255,0.3);
    }

    input:focus{
        outline: none;
        box-shadow: inset 4px 4px 6px 0 rgba(0, 0, 0, 0.2), 
            inset -3px -4px 6px 0 rgba(255, 255, 255, 0.3), 
            0 0 0px 1.7px #d3a392;
    }

    input[type=range] {
        width: 100%;
        margin: 4.05px 0;
        background-color: transparent;
        -webkit-appearance: none;
        box-shadow: none;
      }
      input[type=range]:focus {
        outline: none;
        box-shadow: none;
      }
      input[type=range]::-webkit-slider-runnable-track {
        background: var(--background-light);
        border: 0;
        border-radius: 25px;
        width: 100%;
        height: 1.5em;
        cursor: pointer;
        box-shadow: inset 4px 4px 6px 0 rgba(0, 0, 0, 0.2), 
        inset -3px -4px 6px 0 rgba(255, 255, 255, 0.3);
      }
      input[type=range]::-webkit-slider-thumb {
        margin-top: -4.05px;
        width: 2em;
        height: 2em;
        background: var(--link);
        border: 0;
        border-radius: 2em;
        cursor: pointer;
        -webkit-appearance: none;
        box-shadow: 1.5px 1.5px 4px -1px rgba(0,0,0,0.6), -1px -1px 2px 0px rgba(255,255,255,0.5);
      }
      input[type=range]:focus::-webkit-slider-runnable-track {
        background: var(--background-light);
      }
      input[type=range]::-moz-range-track {
        background: var(--background-light);
        border: 0;
        border-radius: 25px;
        width: 100%;
        height: 1.5em;
        cursor: pointer;
        box-shadow: inset 4px 4px 6px 0 rgba(0, 0, 0, 0.2), 
        inset -3px -4px 6px 0 rgba(255, 255, 255, 0.3);
      }
      input[type=range]::-moz-range-thumb {
        width: 2em;
        height: 2em;
        background: var(--link);
        border: 0;
        border-radius: 2em;
        cursor: pointer;
        box-shadow: 1.5px 1.5px 4px -1px rgba(0,0,0,0.6), -1px -1px 2px 0px rgba(255,255,255,0.5);
      }
      input[type=range]::-ms-track {
        background: transparent;
        border-color: transparent;
        border-width: 9.65px 0;
        color: transparent;
        width: 100%;
        height: 1.5em;
        cursor: pointer;
        box-shadow: inset 4px 4px 6px 0 rgba(0, 0, 0, 0.2), 
        inset -3px -4px 6px 0 rgba(255, 255, 255, 0.3);
      }
      input[type=range]::-ms-fill-lower {
        background: var(--background-light);
        border: 0;
        border-radius: 50px;
      }
      input[type=range]::-ms-fill-upper {
        background: var(--background-light);
        border: 0;
        border-radius: 50px;
      }
      input[type=range]::-ms-thumb {
        width: 2em;
        height: 2em;
        background: var(--link);
        border: 0;
        border-radius: 2em;
        cursor: pointer;
        margin-top: 0px;
        box-shadow: 1.5px 1.5px 4px -1px rgba(0,0,0,0.6), -1px -1px 2px 0px rgba(255,255,255,0.5);
      }
      input[type=range]:focus::-ms-fill-lower {
        background: var(--background-light);
      }
      input[type=range]:focus::-ms-fill-upper {
        background: var(--background-light);
      }
     
      @supports (-ms-ime-align:auto) {
        input[type=range] {
          margin: 0;
        }
      }

`
