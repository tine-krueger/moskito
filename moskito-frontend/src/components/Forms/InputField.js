import styled from 'styled-components/macro'
import PropTypes from 'prop-types'

InputField.propTypes = {
    type: PropTypes.string.isRequired,
    name: PropTypes.string,
    value: PropTypes.node,
    onChange: PropTypes.func,
    placeholder: PropTypes.node,
    children: PropTypes.string,
    marginBottom: PropTypes.string
}

export default function InputField({type, name, value, onChange, placeholder, children, marginBottom}) {
    return (
        <Label data-testid='input-field' lessMargin={marginBottom}>
            <h3>{children}</h3>
            <input type={type} name={name} value={value} onChange={onChange} placeholder={placeholder} />
        </Label>
    )
}

const Label = styled.label `
    grid-column: 1 / -1;
    width: 100%;
    text-align: center;
    margin: 0 auto 2em;
    margin-bottom: ${(props) => props.lessMargin === 'sm' ? '.5em' : '1em'};
    h3 {
        text-align: left;
        padding-left: .5em;
        margin: 0 0 1em;
    }

    input {
        width: 100%;
        padding: 1em;
        height: 3em;
        font-size: 1em;
        border: none;
        border-radius: 10px;
        color: #6b717e;
        background-color: #d8e6e4;
        box-shadow: inset 4px 4px 6px 0 rgba(0,0,0,0.2), inset -3px -4px 6px 0 rgba(255,255,255,0.3);
    }
    input:focus{
        outline: none;
        box-shadow: inset 4px 4px 6px 0 rgba(0, 0, 0, 0.2), 
                    inset -3px -4px 6px 0 rgba(255, 255, 255, 0.3),
                    0 0 0px 1.7px #d3a392;
    }
`
