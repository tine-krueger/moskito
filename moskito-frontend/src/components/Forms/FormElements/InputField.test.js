import { render } from '@testing-library/react'
import 'jest-styled-components'
import InputField from './InputField'


const onChange = jest.fn()
const component = <InputField 
type='text' 
name='address' 
value='anystreet' 
onChange={onChange} 
placeholder='name of street' 
children='Address'
/>

const componentMargin = <InputField 
type='text' 
name='address' 
value='anystreet' 
onChange={onChange} 
placeholder='name of street' 
children='Address'
marginBottom='sm'
/>

describe('InputField', () => {
    it('renders correctly', () => {
        const { container } = render(component)
        expect(container).toMatchSnapshot()
    })

    it('shows right label', () => {
        const { queryByLabelText } = render(component)
        expect(queryByLabelText('Address')).toBeTruthy()
        expect(queryByLabelText('address')).toBeFalsy()
    })

    it('with no margin-prop has margin bottom of 1em', () => {
        const { getByTestId } = render(component)
        expect(getByTestId('input-field')).toHaveStyle('margin-bottom: 1em')
    })

    it('with margin-prop has margin bottom of .5em', () => {
        const { getByTestId } = render(componentMargin)
        expect(getByTestId('input-field')).toHaveStyle('margin-bottom: .5em')
    })
})