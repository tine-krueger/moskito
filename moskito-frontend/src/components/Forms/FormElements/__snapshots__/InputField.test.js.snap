// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField renders correctly 1`] = `
.c0 {
  grid-column: 1 / -1;
  width: 100%;
  text-align: center;
  margin: 0 auto 2em;
  margin-bottom: 1em;
}

.c0 h3 {
  text-align: left;
  padding-left: .5em;
  margin: 0 0 1em;
}

.c0 input {
  width: 100%;
  padding: 1em;
  height: 3em;
  font-size: 1em;
  border: none;
  border-radius: 10px;
  color: #6b717e;
  background-color: #d8e6e4;
  box-shadow: inset 4px 4px 6px 0 rgba(0,0,0,0.2),inset -3px -4px 6px 0 rgba(255,255,255,0.3);
}

.c0 input:focus {
  outline: none;
  box-shadow: inset 4px 4px 6px 0 rgba(0,0,0,0.2),inset -3px -4px 6px 0 rgba(255,255,255,0.3),0 0 0px 1.7px #d3a392;
}

<div>
  <label
    class="c0"
    data-testid="input-field"
  >
    <h3>
      Address
    </h3>
    <input
      name="address"
      placeholder="name of street"
      type="text"
      value="anystreet"
    />
  </label>
</div>
`;
